gammapy/data/observations.py

```diff
@@ -715,6 +716,23 @@ def group_by_label(self, labels):
             obs_groups[f"group_{label}"] = observations
         return obs_groups
 
+    @classmethod
+    def from_stack(cls, observations_list):
+        """Create a new `Observations` instance by concatenating a list of `Observations` objects.
+
+        Parameters
+        ----------
+        observations_list : list of `~gammapy.data.Observations`
+            The list of `Observations` to concatenate.
+
+        Returns
+        -------
+        observations : `~gammapy.data.Observations`
+            The `Observations` object resulting from the concatenation of all the `Observations` in `observation_list`.
+        """
+        obs = itertools.chain(*observations_list)
```

Comment:
I would propose to add a function `duplication_check` before the concatenation, because this function is soooo dangerous.
Today, only the obs_id can be checked. Later on,  when we will have metadata, better checks  could be added.